sdk/go/common/workspace/plugins.go

```diff
@@ -317,14 +318,148 @@ func (info PluginInfo) Download() (io.ReadCloser, int64, error) {
 		if err == nil {
 			return resp, length, nil
 		}
+		logging.V(1).Infof("cannot find plugin on github.com/pulumi/pulumi-%s/releases", info.Name)
+
+		resp, length, err = getPluginFromPrivateGitHubResponse(info, opSy, arch)
+		if err == nil {
+			return resp, length, nil
+		}
+		logging.V(1).Infof("cannot find plugin %s on private GitHub releases: %s", info.Name, err.Error())
 
 		// we threw an error talking to GitHub so lets fallback to get.pulumi.com for the provider
-		logging.V(1).Infof("cannot find plugin on github.com/pulumi/pulumi-%s/releases", info.Name)
 	}
 
 	return getPluginResponse(buildPulumiHostedPluginURL(info.Kind, info.Name, info.Version, opSy, arch))
 }
 
+func getPluginFromPrivateGitHubResponse(info PluginInfo, opSy, arch string) (io.ReadCloser, int64, error) {
+	assetName := fmt.Sprintf("pulumi-%s-%s-v%s-%s-%s.tar.gz",
+		info.Kind, info.Name, info.Version.String(), opSy, arch)
+	resp, length, err := getPluginPrivateGitHubReleaseAPIResponse(info)
```

Comment:
Why do you need to hit the releases endpoint? For pulumi plugins we expect them at "https://github.com/pulumi/pulumi-%s/releases/download/v%s/%s" and I'd assume private plugins could be the same but under a different org so "https://github.com/%s/pulumi-%s/releases/download/v%s/%s",

Reply:
That is the browser link and requires the client to be authenticated as a browser, which is not documented. The documented way of downloading from private releases is to use GitHub's releases `API` and using Basic or OAuth authentication.